server/fleet/geoip.go

```diff
@@ -0,0 +1,79 @@
+package fleet
+
+import (
+	"context"
+	"errors"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/oschwald/geoip2-golang"
+	"net"
+)
+
+var notCityDBError = geoip2.InvalidMethodError{}
+
+type GeoLocation struct {
+	CountryISO string    `json:"country_iso"`
+	CityName   string    `json:"city_name"`
+	Geometry   *Geometry `json:"geometry,omitempty"`
+}
+
+type Geometry struct {
+	Type        string    `json:"type"`
+	Coordinates []float64 `json:"coordinates"`
+}
+
+type GeoIP interface {
+	Lookup(ctx context.Context, ip string) *GeoLocation
+}
+
+type MaxMindGeoIP struct {
+	reader *geoip2.Reader
+	l      log.Logger
+}
+
+type NoOpGeoIP struct{}
+
+func (n *NoOpGeoIP) Lookup(ctx context.Context, ip string) *GeoLocation {
+	return nil
+}
+
+func NewMaxMindGeoIP(logger log.Logger, path string) GeoIP {
+	r, err := geoip2.Open(path)
+	if err != nil {
+		return &NoOpGeoIP{}
```

Comment:
Mhm, we should let the caller make this decision:
`func NewMaxMindGeoIP(logger log.Logger, path string) (GeoIP, error)`
The caller can log the error and then resort to `NoOpGeoIP{}`.